Provide helpers for sanity checking.

freqlabs opened the pull request:
Many functions throughout lib/ repeat sanity checks on $name and $dataset. The functions __require_name and __require_dataset are intended to replace these repetitive tasks.

Update __set_jail_prop to use the new sanity checking helpers.

Commit:
Provide helpers for sanity checking.
Update __set_jail_prop to use sanity checking helpers.

Commit:
Update ioc-common to use sanity check helpers.

freqlabs commented:
Meant to hold off on that second commit, but oops there it is now.

skarekrow commented on lib/ioc-common:
```diff
@@ -10,17 +10,25 @@ __export_props () {
         exit 1
     fi
 
+    echo "$_name"
```
Is there a reason you have this here? Or is that just debug you forgot in?

skarekrow commented on lib/ioc-common:
```diff
@@ -33,6 +41,22 @@ __set_jail_prop () {
         exit 1
     fi
 
+    echo "$_dataset"
```
Same as above.

freqlabs commented:
That's effectively the return statement for the functions. They don't modify variables outside their scope, instead they just echo their output. That's why they are used as _name="$(__require_name $1)" and _dataset="$(__require_dataset $_name)". The helpers perform the necessary checks, returning the value to be stored if the checks pass, otherwise displaying an error message and exiting.

pannon commented:
@freqlabs definitely supportive on the idea. Just need some time to test.

Commit:
Correct use of quotations.

Updates the changes for the __requires_* helpers to so that the
arguments are properly quoted when calling the functions.
